ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java

```diff
@@ -613,14 +611,7 @@ private void performHardDeletes(List<MessageInfo> messageInfoList) throws StoreE
         }
         logWriteInfo.logSegment.writeFrom(logWriteInfo.channel, logWriteInfo.offset, logWriteInfo.size);
         metrics.hardDeleteDoneCount.inc(1);
-        throttler.maybeThrottle(logWriteInfo.size);
-      }
-    } catch (InterruptedException e) {
-      if (enabled.get()) {
-        // We throw here because we do not want the tokens to be updated.
-        throw new StoreException("Got interrupted during hard deletes", StoreErrorCodes.Unknown_Error);
-      } else {
-        throw new StoreException("Got interrupted as store is shutting down", StoreErrorCodes.Store_Shutting_Down);
+        diskIOScheduler.getSlice(DiskManager.CLEANUP_OPS_JOB_NAME, DiskManager.CLEANUP_OPS_JOB_NAME, logWriteInfo.size);
```

Comment:
To your first point, it is expected that the class should know what its job type and job id should be. 
To your second, how is having a static function different from having this static variable?